### ISSUE TITLE
Support modern browser runtime?

### ISSUE THREAD

**Issue by blakegong** [roryrjb/md5-file#11]
Hey Rory,

Thanks for this great repo!

I am wondering whether you would like to add modern browser support? Maybe utilising `FileReader` API to handle `fs` work?

It would be great to see that happen. I can contribute with some PR too :)

Best,
Blake

**Comment by roryrjb** [comment]
@blakegong on one hand I wouldn't want this module to get too complicated, but on the other, having a _one stop shop_ for MD5 check-summing in JavaScript might be really handy. Go for it.

**Comment by blakegong** [comment]
Thanks for the blessing. Will try my best :)

**Comment by blakegong** [comment]
Hi Rory,

Just a small update here. I didn't find enough time to do it last weekend. But managed to write a prototype using https://www.npmjs.com/package/md5 that works on the browser. Based on that experiment, there are two things that I wanna discuss here.

1. Since `crypto` package only exists in `node`, in order to make it work in browser, would you prefer to write md5 calculation logic by hand, or simply use the aforementioned library?
1. Personally I intend to keep the current API. Problem is, browser version of this library will introduce two additional files `browser.js` and `promise-browser.js` in order to make importing from `md5-file` and `md5-file/promise` work (exporting different code based on different `node.env` would make file size bigger. e.g.: node does not need all the new logics I am gonna put in this library). Thus I am hoping to get your approval to change the file structure, placing current `index.js` and `promise.js` into `/node/` or `/src/node/`, and the same for browser code. In this way, better code splitting and clear file structure should be achieved.

**Comment by roryrjb** [comment]
@blakegong I wouldn't worry too much about the increased file size, the `md5` package and its dependencies are tiny compared to if you used __Browserify__ on the `md5-file` module itself which results in a file around 500KB. Actually talking about __Browserify__, I think we have to consider what happens when someone will use that method of incorporating `md5-file` into the browser vs. using a `<script>` tag for example.

**Comment by blakegong** [comment]
Agreed. I myself mainly write React everyday so I guess I was underestimating the work required to make this package truly *work* in browser.

Guess we should support at least 3 different types of import?

- `<script>` tag
- `require()` as you have already done here
- ES6 `import` syntax

**Comment by roryrjb** [comment]
@blakegong yeah sounds good
